d2/src/main/proto/XdsD2.proto

```diff
@@ -40,5 +40,55 @@ message D2SymlinkNode {
 }
 
 message D2NodeMap {
+  // Deprecated in favor of D2UriMap
+  option deprecated = true;
   map<string, D2Node> nodes = 1;
 }
+
+// D2URI is a proto representation of com.linkedin.d2.balancer.properties.UriProperties. Note that a D2 UriProperties is
+// actually an announcement from a single machine for potentially multiple URIs. This class is a misnomer but it seems
+// important to be consistent with the old naming strategy as this new proto bridges the old type. The string keys in
+// the maps of the message below will be the URIs of this announcement. Here is a sample ZK announcement for additional
+// clarity:
+//	{
+//		"weights": {
+//			"https://foo.stg.linkedin.com:18792/Toki/resources": 1.0
+//		},
```

Comment:
why we need "weights" field? from this example,  partitionDesc seems like a superset of weights?

Reply:
I might have been confused by this example:).  as an example of D2URI object, it should not have separate weight field?

Reply:
Yes but it's a legacy feature. I linked the full contents of a URI node but "weights" is ignored here, it's actually mentioned in the comment on `partition_desc`